sdk/src/message.rs

```diff
@@ -70,34 +70,67 @@ fn get_program_ids(instructions: &[Instruction]) -> Vec<Pubkey> {
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
 pub struct Message {
-    #[serde(skip)]
     pub num_signatures: u8,
+
+    /// All the account keys used by this transaction
     #[serde(with = "short_vec")]
     pub account_keys: Vec<Pubkey>,
+
+    /// The id of a recent ledger entry.
     pub recent_blockhash: Hash,
+
+    /// The number of lamports paid for processing and storing of this transaction.
     pub fee: u64,
+
+    /// All the program id keys used to execute this transaction's instructions
     #[serde(with = "short_vec")]
     pub program_ids: Vec<Pubkey>,
+
+    /// Programs that will be executed in sequence and committed in one atomic transaction if all
+    /// succeed.
     #[serde(with = "short_vec")]
     pub instructions: Vec<CompiledInstruction>,
 }
 
 impl Message {
+    pub fn new_with_compiled_instructions(
+        num_signatures: u8,
+        account_keys: Vec<Pubkey>,
+        recent_blockhash: Hash,
+        fee: u64,
+        program_ids: Vec<Pubkey>,
+        instructions: Vec<CompiledInstruction>,
+    ) -> Self {
+        Self {
+            num_signatures,
+            account_keys,
+            recent_blockhash,
+            fee,
+            program_ids,
+            instructions,
+        }
+    }
+
```

Comment:
The following comment has a typo, but I think it's also incorrect now. It doesn't return a transaction.

Reply:
Thanks, that was copied from Transaction. I deleted it.